### ISSUE TITLE
Differen values on "get_latest_training_loss" in Word2Vec

### ISSUE THREAD

**Issue by yu45020** [RaRe-Technologies/gensim#2039]
I train a Word2Vec using the standard way described in the API document. To get loss of each epoch, I also train the same model with decreasing learning rate per epoch. 
Based on the [source code](https://github.com/RaRe-Technologies/gensim/blob/916e423a1654d87646fc522d8862a81c7a7cb4fc/gensim/models/base_any2vec.py#L579), the internal learning rate is decreasing linearly per epoch, so I do the same thing as shown below. The last loss values are quite different. 

* Standard way

 ```
word2vec_params = {
    'sg': 0,  # 0 ： CBOW； 1 : skip-gram
    "size": 300,
    "alpha": 0.5,
    "min_alpha": 0.001,
    'window':10,
    'min_count': 1,
    'seed': 1,
    "workers": 4,
    "negative": 0,
    "hs": 1,  # 0: negative sampling, 1:hierarchical  softmax
    'compute_loss': True,
    'iter': 10,
    'cbow_mean':1,
}

model = Word2Vec(**word2vec_params)
model.build_vocab(sentences)

trained_word_count, raw_word_count = model.train(sentences, compute_loss=True,
                                                 total_examples=model.corpus_count,
                                                 epochs=model.epochs)
model.get_latest_training_loss()
# return: 49245.1171875
```

* Per epoch (type 1) 

start alpha = end_alpha
```
model = Word2Vec(**word2vec_params)
model.build_vocab(sentences)
losses = []
learning_rate = 0.5
step_size = (0.5 - 0.001) / 10

for i in range(10):
    trained_word_count, raw_word_count = model.train(sentences, compute_loss=True,
                                                     start_alpha=learning_rate,
                                                     end_alpha=learning_rate,
                                                     total_examples=model.corpus_count,
                                                     epochs=1)
    loss = model.get_latest_training_loss()
    losses.append(loss)
    print(i, loss, learning_rate)
    learning_rate -= step_size
```
Each epoch loss is 
```
[58784.37109375,
 2431.101806640625,
 1676.5684814453125,
 1455.739990234375,
 1259.876953125,
 1206.9366455078125,
 1156.148193359375,
 1021.84326171875,
 1008.6217041015625,
 887.54541015625]
```

* Per epoch (type 2) 

start_alpha - end_alpha = step_size
```
model = Word2Vec(**word2vec_params)
model.build_vocab(sentences)
losses = []
learning_rate = 0.5
step_size = (0.5 - 0.001) / 10

for i in range(10):
    end_lr = learning_rate - step_size
    trained_word_count, raw_word_count = model.train(sentences, compute_loss=True,
                                                     start_alpha=learning_rate,
                                                     end_alpha=end_lr,
                                                     total_examples=model.corpus_count,
                                                     epochs=1)
    loss = model.get_latest_training_loss()
    losses.append(loss)
    print(i, loss, learning_rate)
    learning_rate -= step_size
```
Each epoch loss is 
```
[35804.48828125,
 2652.100830078125,
 1748.483642578125,
 1293.23583984375,
 1150.828369140625,
 918.9574584960938,
 951.2122192382812,
 871.8506469726562,
 888.2496337890625,
 846.38427734375]
```

**Comment by menshikh-iv** [comment]
Hello @yu45020, training is non-deterministic in this case (because you use multiple workers), for this reason, you receive different values even if you run same code twice (see Gordon answer https://groups.google.com/forum/#!searchin/gensim/word2vec$20workers%7Csort:date/gensim/t86BZUR8qa8/TlLsbHOpBgAJ)
